src/include/cursor.h

```diff
@@ -287,6 +287,16 @@ struct __wt_cursor_hs {
 
     WT_CURSOR *file_cursor; /* Queries of regular history store data */
     WT_TIME_WINDOW time_window;
+    uint32_t btree_id;
+    WT_ITEM ds_key;
+
+    /* AUTOMATIC FLAG VALUE GENERATION START */
+#define WT_HSC_BTREE_ID_SET 0x1u
```

Comment:
I guess naming can be improved. How about WT_HS_CUR_* prefix?

Reply:
I'm following the convention of metadata cursor. Yes, I think we can improve it.